src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/service/realtime.py

```diff
@@ -777,25 +872,39 @@ def realtime_service_view(service_name=None, verb=False, context=cli_context):
             print('[Local mode] Error: Misconfigured container. Cannot determine scoring port.')
             return
     else:
-        if context.acs_agent_url is not None:
-            scoring_url = 'http://' + context.acs_agent_url + ':9091/score'
-            sample_url = 'http://' + context.acs_agent_url + ':9091/sample'
-            headers = {'Content-Type': 'application/json', 'X-Marathon-App-Id': "/{}".format(service_name)}
-            usage_headers.append('-H "X-Marathon-App-Id:/{}"'.format(service_name))
+        if context.env_is_k8s:
+            try:
+                fe_url = get_k8s_frontend_url()
+            except ApiException:
+                return
+            scoring_url = fe_url + service_name + '/score'
+            sample_url = fe_url + service_name + '/sample'
+            headers = {'Content-Type': 'application/json'}
         else:
-            print('Unable to determine ACS Agent URL. '
-                  'Please ensure that AML_ACS_AGENT environment variable is set.')
-            return
+            if context.acs_agent_url is not None:
+                scoring_url = 'http://' + context.acs_agent_url + ':9091/score'
+                sample_url = 'http://' + context.acs_agent_url + ':9091/sample'
+                headers = {'Content-Type': 'application/json', 'X-Marathon-App-Id': "/{}".format(service_name)}
+                usage_headers.append('-H "X-Marathon-App-Id:/{}"'.format(service_name))
+            else:
+                print('Unable to determine ACS Agent URL. '
+                      'Please ensure that AML_ACS_AGENT environment variable is set.')
+                return
 
     service_sample_data = get_sample_data(sample_url, headers, verbose)
     sample_data = '{{"input":"{}"}}'.format(
         service_sample_data if service_sample_data is not None else default_sample_data)
-    print('Usage:')
-    print('  az ml  : az ml service run realtime -n {} [-d \'{}\']'.format(service_name, sample_data))
-    print('  curl : curl -X POST {} --data \'{}\' {}'.format(' '.join(usage_headers), sample_data, scoring_url))
+    if num_services:
+        print('Usage:')
+        print('  az ml  : az ml service run realtime -n {} [-d \'{}\']'.format(service_name, sample_data))
+        print('  curl : curl -X POST {} --data \'{}\' {}'.format(' '.join(usage_headers), sample_data, scoring_url))
 
 
 def realtime_service_list(service_name=None, verb=False, context=cli_context):
```

Comment:
realtime_service_list gets called by az itself. Check out commands.py. It doesn't show up for resharper/pycharm because it's just a string.